boot/zephyr/nrf_cleanup.c

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <hal/nrf_rtc.h>
+#include <hal/nrf_uarte.h>
+#include <hal/nrf_clock.h>
+
+#if CONFIG_HAS_HW_NRF_RTC0 || CONFIG_HAS_HW_NRF_RTC1 || CONFIG_HAS_HW_NRF_RTC2
+static inline void nrf_cleanup_rtc(NRF_RTC_Type * rtc_reg)
+{
+    nrf_rtc_task_trigger(rtc_reg, NRF_RTC_TASK_STOP);
+    nrf_rtc_event_disable(rtc_reg, 0xFFFFFFFF);
+    nrf_rtc_int_disable(rtc_reg, 0xFFFFFFFF);
+}
+#endif
+
+static void nrf_cleanup_clock(void)
+{
+    nrf_clock_int_disable(NRF_CLOCK, 0xFFFFFFFF);
+}
+
+void nrf_cleanup_peripheral(void)
+{
+#if CONFIG_HAS_HW_NRF_RTC0
```

Comment:
Kconfig options `HAS_HW_NRF_*` are supposed to be removed soon. Could you use something different here, like `#if defined(NRF_RTC0)` for example? To avoid updating it in some near future.